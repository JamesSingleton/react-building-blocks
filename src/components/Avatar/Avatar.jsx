import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledAvatar = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${(props) => {
    if (props.size.toLowerCase() === 'large') return '4.5rem';
    if (props.size.toLowerCase() === 'small') return '2.25rem';
    return '3rem';
  }};
  width: ${(props) => {
    if (props.size.toLowerCase() === 'large') return '4.5rem';
    if (props.size.toLowerCase() === 'small') return '2.25rem';
    return '3rem';
  }};
  border-radius: ${(props) => {
    if (props.size.toLowerCase() === 'large') return '2.25rem';
    if (props.size.toLowerCase() === 'small') return '1.3125rem';
    return '1.875rem';
  }};
  background-image: linear-gradient(225deg,#389aec,#4ebcf5);
`;

const StyledName = styled.div`
  color: #ffffff;
  font-size: ${(props) => {
    if (props.size.toLowerCase() === 'large') return '1rem';
    if (props.size.toLowerCase() === 'small') return '0.75rem';
    return '0.875rem';
  }};
  line-height: ${(props) => {
    if (props.size.toLowerCase() === 'large') return '1.5rem';
    if (props.size.toLowerCase() === 'small') return '0.9375rem';
    return '1.125rem';
  }};
`;

const Avatar = ({ name, size, color }) => (
  <StyledAvatar color={color} size={size}>
    <StyledName size={size}>{name}</StyledName>
  </StyledAvatar>
);

Avatar.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large',
  ]),
  color: PropTypes.string.isRequired,
};

Avatar.defaultProps = {
  size: 'medium',
};

export default Avatar;
