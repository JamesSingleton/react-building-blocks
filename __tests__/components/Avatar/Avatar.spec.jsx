import React from 'react';
import { mount } from 'enzyme';
import Avatar from '../../../src/components/Avatar';

describe('Avatar', () => {
  it('should render a small avatar', () => {
    const props = {
      size: 'small',
      name: 'JS',
      color: 'red',
    };
    const renderedAvatar = mount(<Avatar {...props} />);
    expect(renderedAvatar).toMatchSnapshot();
  });
  it('should render a medium avatar', () => {
    const props = {
      name: 'JS',
      color: 'green',
    };
    const renderedAvatar = mount(<Avatar {...props} />);
    expect(renderedAvatar).toMatchSnapshot();
  });
  it('should render a large avatar', () => {
    const props = {
      size: 'large',
      name: 'JS',
      color: 'blue',
    };
    const renderedAvatar = mount(<Avatar {...props} />);
    expect(renderedAvatar).toMatchSnapshot();
  });
  it('should render a purple avatar', () => {
    const props = {
      name: 'JS',
      color: 'purple',
    };
    const renderedAvatar = mount(<Avatar {...props} />);
    expect(renderedAvatar).toMatchSnapshot();
  });
  it('should render a orange avatar', () => {
    const props = {
      name: 'JS',
      color: 'orange',
    };
    const renderedAvatar = mount(<Avatar {...props} />);
    expect(renderedAvatar).toMatchSnapshot();
  });
});
