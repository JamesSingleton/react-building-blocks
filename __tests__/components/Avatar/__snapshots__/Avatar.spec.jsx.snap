// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render a large avatar 1`] = `
<Avatar
  color="blue"
  name="JS"
  size="large"
>
  <styled.div
    color="blue"
    size="large"
  >
    <div
      className="sc-AykKC kA-dXtV"
      color="blue"
      size="large"
    >
      <styled.div
        size="large"
      >
        <div
          className="sc-AykKD fvTGaU"
          size="large"
        >
          JS
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;

exports[`Avatar should render a medium avatar 1`] = `
<Avatar
  color="green"
  name="JS"
  size="medium"
>
  <styled.div
    color="green"
    size="medium"
  >
    <div
      className="sc-AykKC eMLSiC"
      color="green"
      size="medium"
    >
      <styled.div
        size="medium"
      >
        <div
          className="sc-AykKD fzJCPo"
          size="medium"
        >
          JS
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;

exports[`Avatar should render a orange avatar 1`] = `
<Avatar
  color="orange"
  name="JS"
  size="medium"
>
  <styled.div
    color="orange"
    size="medium"
  >
    <div
      className="sc-AykKC kjkDeA"
      color="orange"
      size="medium"
    >
      <styled.div
        size="medium"
      >
        <div
          className="sc-AykKD fzJCPo"
          size="medium"
        >
          JS
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;

exports[`Avatar should render a purple avatar 1`] = `
<Avatar
  color="purple"
  name="JS"
  size="medium"
>
  <styled.div
    color="purple"
    size="medium"
  >
    <div
      className="sc-AykKC bMgnjD"
      color="purple"
      size="medium"
    >
      <styled.div
        size="medium"
      >
        <div
          className="sc-AykKD fzJCPo"
          size="medium"
        >
          JS
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;

exports[`Avatar should render a small avatar 1`] = `
<Avatar
  color="red"
  name="JS"
  size="small"
>
  <styled.div
    color="red"
    size="small"
  >
    <div
      className="sc-AykKC BiPoB"
      color="red"
      size="small"
    >
      <styled.div
        size="small"
      >
        <div
          className="sc-AykKD iWWKhD"
          size="small"
        >
          JS
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;
